### ISSUE TITLE
tables in gmail don't work properly due to newlines

### ISSUE THREAD

**Issue by hchaase** [adam-p/markdown-here#251]
in gmail, *newlines* at end of table rows break the table. See [stackexchange post](https://stackoverflow.com/questions/21125721/markdown-heretables-do-not-display-correctly-upon-toggling-to-markdown)

**Comment by adam-p** [comment]
Can you supply more info? Browser? Site (like Gmail, etc.)? A sample of the Markdown? Screenshots?

**Comment by hchaase** [comment]
Firefox (The version Just before The current release) writing in gmail
pubup. Simple tables. Only worked if i pressede shift and enter at the same
time for New lines.

HC
Den 10/03/2015 00.56 skrev "Adam Pritchard" <notifications@github.com>:

**Comment by adam-p** [comment]
I can't reproduce the problem.

Can you type in a MD table (*without* pressing shift), and then follow the animation instructions here for copying the HTML, then share the HTML here? Thanks.

https://github.com/adam-p/markdown-here/wiki/Troubleshooting#copying-html

**Comment by hchaase** [comment]
here are the original text

mardown | test
-- |---
1 | test
2 | test2


not quite sure if i got the right snip of code, but lets try



<div itacorner="6,7:1,1,0,0" tabindex="1" style="direction: ltr;
min-height: 374px;" role="textbox" g_editable="true" id=":156" class="Am Al
editable LW-avf" hidefocus="true" aria-label="Meddelelsens tekst"
contenteditable="true"><div><div>mardown | test<br>-- |---<br></div>1 |
test<br></div>2 | test2<br clear="all"><div><div><div><div><div
class="gmail_signature"><div dir="ltr"><div><br>Med venlig hilsen<br><br>HC
Haase<br></div>Geograf<br><div><br>Direkte telefon: 3529 4422<br>E-post: <a
href="mailto:hha@fredensborgforsyning.dk" target="_blank">
HCH@fredensborgforsyning.dk</a><br>




<br><span></span><span></span><img src="
https://ci6.googleusercontent.com/proxy/kcJQMvuLilr7AuZFKqSplI_NPPs2VrHRh4SCN6rVoB4pO34JmbJWBpI4VH6_DjgR0s3eidsT8AHsLaQ38NAb1kx6xi7MbJawN-RclUukCsH0ASxFRg=s0-d-e1-ft#http://www.fredensborgforsyning.dk/uploads/images/ffas-logo.png"><br><br>Fredensborg
Forsyning A/S<br>Højvangen 23<br>3480 Fredensborg<br>Telefon: 3529
4400<br>Hjemmeside: <a href="http://www.fredensborgforsyning.dk"
target="_blank">www.fredensborgforsyning.dk</a><br>




</div></div></div></div>
</div></div></div></div>


and here are some of the code for the text

<div><div>mardown | test<br>-- |---<br></div>1 | test<br></div>

Venlig hilsen
HC Haase

P    Please consider your environmental responsibility before printing this
email
このメールを印刷される前に、少しだけ環境のことも考えてみてください。小さな行動が地球を救います。

2015-03-10 16:17 GMT+00:00 Adam Pritchard <notifications@github.com>:

**Comment by adam-p** [comment]
After banging my head against this for a while, I finally figured out the problem. Fixed in commit a0c379b, but I won't be putting it in the next release because I need to test it a lot more (and I plan on putting out a release soon).

Here's the offending HTML, pretty-printed.

```html
<div>
  <div>
    mardown | test
    <br>
    -- |---
    <br>
  </div>
  1 | test
  <br>
</div>
2 | test2
```

You can see that the structure has gotten kind of messy, probably due to deleting and re-typing. (Which is when I've noticed newline problems in the past.)

The MDH failure is occurring because: A `<br>` as the last child of a `<div>`, like so...

```
...
  <br>
</div>
...
```

...Does not add an additional (visual) newline. But MDH's HTML-to-text processor treated it as if it did. So you can see from your HTML above that there would be too many newlines and the table structure would be broken.

I fixed it by tweaking the HTML-to-text parsing. As I said, this is kind of risky, so I'm going to leave it unreleased for a while so I can try it out. (The tests pass, though. So... probably okay!)

Thanks for your help, @hchaase.
